icu4c/source/test/intltest/caltest.cpp

```diff
@@ -5628,6 +5630,35 @@ void CalendarTest::TestFirstDayOfWeek() {
 
     verifyFirstDayOfWeek("zxx", UCAL_MONDAY);
 }
+
+void CalendarTest::TestChineseCalendarComputeMonthStart() {  // ICU-22639
+    UErrorCode status = U_ZERO_ERROR;
+
+    // An extended year for which hasLeapMonthBetweenWinterSolstices is true.
+    constexpr int32_t eyear = 4643;
+    constexpr int32_t monthStart = 2453764;
+
+    LocalPointer<Calendar> calendar(
+        Calendar::createInstance(Locale("en_US@calendar=chinese"), status),
+        status);
+    if (failure(status, "Calendar::createInstance")) return;
+
+    // This test case is a friend of ChineseCalendar and may access internals.
+    const ChineseCalendar& chinese =
+        *dynamic_cast<ChineseCalendar*>(calendar.getAlias());
+
+    // The initial value of hasLeapMonthBetweenWinterSolstices should be false.
+    assertFalse("hasLeapMonthBetweenWinterSolstices [#1]",
+                chinese.hasLeapMonthBetweenWinterSolstices);
```

Comment:
Is this assertion going to be broken by the passage of time?

Reply:
The `ChineseCalendar` constructor is supposed to always set this to `false`, regardless of when the constructor is called, so no, this should not be affected by the passage of time.

Reply:
Ah right, this is only going to be set when the calendar is completed.